scipy/ndimage/_ni_support.py

```diff
@@ -79,11 +79,3 @@ def _get_output(output, input, shape=None):
     elif output.shape != shape:
         raise RuntimeError("output shape not correct")
     return output
-
-
-def _check_axis(axis, rank):
-    if axis < 0:
-        axis += rank
-    if axis < 0 or axis >= rank:
-        raise ValueError('invalid axis')
```

Comment:
Note that this is technically a compatibility break, as users may be catching the `ValueError` specifically, which will instead turn into an `AxisError` after this PR. I expect that shouldn't be too problematic in practice, but it's worth thinking about.

Reply:
`set(AxisError.__bases__) == {ValueError, IndexError}` to avoid exactly this issue.